docs/topics/secured-mission-vertx-adapter.adoc

```diff
@@ -0,0 +1,40 @@
+= SSO Adapter Configuration
+
+The SSO adapter is the _client side_, or client to the SSO server, component that enforces security on the web resources. In this specific case, it is the greeting service.
+
+.Enacting security
+[source,java,options="nowrap",subs="attributes+"]
+----
+router.route("/greeting")         <1>
+  .handler(JWTAuthHandler.create( <2>
+    JWTAuth.create(vertx,         <3>
+      config)));                  <4>
+
+----
+
+<1> Locate the HTTP route to secure.
+<2> Instantiate a new JWT security handler.
+<3> The authorization enforcer is created.
+<4> The configuration to the enforcer.
+
+The enforcer is configured using JSON with the exact same format keycloak exports from the configuration console. In this case the JSON file is reconstructed from the deployment environment variables as:
+
+[source,bash,options="nowrap",subs="attributes+"]
+----
+JsonObject config = new JsonObject()
+  	.put("realm", System.getenv("REALM")) <1>
+  	.put("public-key", System.getenv("REALM_PUBLIC_KEY")) <3>
+  	.put("auth-server-url", System.getenv("SSO_AUTH_SERVER_URL")) <4>
+  	.put("ssl-required", "external")
+  	.put("resource", System.getenv("CLIENT_ID")) <2>
+  	.put("credentials", new JsonObject()
+    	.put("secret", System.getenv("SECRET")))
+  	// since we're consuming keycloak JWTs we need to locate the permission claims in the token
+  	.put("permissionsClaimKey", "realm_access/roles"); <5>
```

Comment:
Why 1, 3, 4, 2, 5?

Reply:
In the source, they are just identifiers. AsciiDoc replaces the numbers so that it's 1, 2, 3, 4, 5.

Reply:
Ah interesting :-)